### ISSUE TITLE
Keycard authentication devices are pretty dumb

### ISSUE THREAD

**Issue by gbasood** [d3athrow/vgstation13#4473]
Making a new issue for this so that discussion in the other issue isn't taken off topic.

I'm talking about the devices that are currently only good for setting red alert status.
They should either be given more features or have their functions moved to the head's PDA modules.
### What I wrote before:

> A potential feature for keycard authentication devices (which have a very silly and useless name) would be allowing, for instance, the CMO to quarantine medbay. Simple implementation would be just activating every medical area's fire alarms, complex would be 'You can't open these doors unless you have medical access, otherwise you have to do some incredibly simple hacking first.'
> 
> Alternatively the things could be removed and stuck onto PDA modules. More functional that way than they are now anyways.

**Comment by Probe1** [comment]
I'm not keen at all on the idea of moving their functions to PDA because the intent was to put static objects in the heads offices that they had to travel to, and swipe their head IDs at simultaneously, to get something major to happen.  Moving it to the PDA negates the entire purpose of the object and lets people declare code red anywhere. 

What I suggested (in the thread and in the previous issue) was that the card readers are ripe for content that can do powerful and interesting things.  I like your idea of the CMO having the ability to lock down medbay.  That's a neat idea.    Having the ability for a head to repeat access requirements temporarily to let security, or virus victims or whatever, into their department, is a neat idea that lets a head swipe by himself instead of coordinating since its something to effect his department only.  It also fleshes out the card readers that have a single purpose (code red, no one ever does maint access ever and why would they)

I feel like the repeal access requirements idea *must* be temporary or else one cute clown can go around with the spare and repeal access to everywhere forever and people can't do much.  Having set on a 10 minute/X minute timer is a good way of making it work without it being all powerful.  

Likewise for quarantine measures.  The buttons in Engineering and Research could be moved to the card reader.  There could be a unique quarantine one for medbay that repeals access internally and bolts doors externally so people can't leave but they can move around medbay very easily.

I think the card reader has potential to do a lot of cool stuff and it's exciting to think about.

**Comment by Shadowmech88** [comment]
I'd also be open to more functionality for them. For example, Science's biohazard lockdown could be moved to the card reader instead of just a button on the RD's desk that isn't even access-restricted.

**Comment by Intigracy** [comment]
@clusterfack milestone

#9999 for my own reference

**Comment by nuklearcellphoneg** [comment]
Yes please.

> virus
> lol no quarantine
